be/src/runtime/fragment_mgr.cpp

```diff
@@ -483,6 +486,14 @@ void FragmentMgr::_exec_actual(std::shared_ptr<PlanFragmentExecutor> fragment_ex
                             std::chrono::system_clock::now().time_since_epoch())
                             .count();
         std::lock_guard<std::mutex> lock(_lock);
+
+        if (query_ctx->enable_profile()) {
```

Comment:
add this logic to PlanFragmentExecutor's close method like PipelineXFragmentContext

Reply:
done